### ISSUE TITLE
[Bug] Upgrade to gRPC 1.46 once it's released (which fixes fork support)

### ISSUE THREAD

**Issue by vakker** [ray-project/ray#22518]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Ray Core

### What happened + What you expected to happen

With the new 1.44.0 grpcio package there's an error message:
```
E0219 11:51:45.606983573  985146 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies
E0219 11:51:45.717404158  985146 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies
2022-02-19 11:51:46,630 INFO services.py:1338 -- View the Ray dashboard at http://127.0.0.1:8265
E0219 11:51:46.633503280  985146 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies
E0219 11:51:46.638523511  985146 fork_posix.cc:70]           Fork support is only compatible with the epoll1 and poll polling strategies
```

Repro script:
```python
import ray

ray.init()
```

### Versions / Dependencies

Tested with Ray ~1.9 and ~1.10 on an Ubuntu 20.04.
This was introduced by grpcio 1.44 (tested it with 1.43 and 1.42, those don't have this issue).

### Reproduction script

See above.

### Anything else

The ERROR log was changed to INFO in this PR: https://github.com/grpc/grpc/pull/17341
Then it was changed back in this PR: https://github.com/grpc/grpc/pull/28566

I don't know if this is an upstream issue or this has any implications to Ray that was masked previously by the INFO log.

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

**Comment by rkooo567** [comment]
cc @iycheng seems related to test output

**Comment by rkooo567** [comment]
Seems like we can easily solve it by setting `GRPC_FORK_SUPPORT_ENABLED =0` in python? or `GRPC_FORK_SUPPORT_ENABLED =1 ` in core

```
Yes, fork+exec is always supported. T[here](https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/fork_posix.cc#L56)'s a mismatch between the default values of GRPC_FORK_SUPPORT_ENABLED in core, where it defaults to true, and in the Python layer, where it defaults to false. There is no signal in core that could let us distinguish beforehand between fork without exec and fork with exec. This is why the earlier error log message (here) is not "log spam" by default. I am fine changing the "Fork support is only compatible..." log statement in fork_posix.cc to INFO level, but we still want an error to surface to Python users that have explicitly set GRPC_FORK_SUPPORT_ENABLED=1. This can probably be done by checking grpc_get_poll_strategy_name() in fork_posix.pyx.pxi.
```

**Comment by rkooo567** [comment]
It seems like the fork support is to allow for clients to be usable after forked by multi processing. I believe it is not the use case from Ray core, so it might be safe to disable this from the core configuration?

**Comment by nikhil-sthalekar** [comment]
Not sure if this is helpful, but I was getting this error in a workflow job this past Sunday and Yesterday (today is feb 22), yet it seemed to resolve itself today.

**Comment by HeyItsBethany3** [comment]
@rkooo567 I'm still experiencing this issue - what is the solution? 
I have tried setting GRPC_FORK_SUPPORT_ENABLED as an environmental variable using `os.environ["GRPC_FORK_SUPPORT_ENABLED"]="0"` in python and `export GRPC_FORK_SUPPORT_ENABLED=0` in the terminal but I'm still getting these error messages.

**Comment by driedler** [comment]
This is an absolutely god awful work-around, but it works:

Basically, it intercepts the stdout out stream and drops the grpc error msg. Use at your own risk:

```
stdout_fileno = sys.stdout.fileno()
stdout_save = os.dup(stdout_fileno)
stdout_pipe = os.pipe()
os.dup2(stdout_pipe[1], stdout_fileno)
os.close(stdout_pipe[1])

def drain_pipe():
    EOL = '\n'.encode('utf-8')[0]
    GRPC_ERR_MSG = 'Fork support is only compatible'.encode('utf-8')

    def readline():
        line = b''
        while True:
            line += os.read(stdout_pipe[0], 1)
            if line[-1] == EOL:
                break 
        return line


    while True:
        line = readline()
        if GRPC_ERR_MSG in line:
            continue 
        os.write(stdout_save, line)

t = threading.Thread(target=drain_pipe)
t.start()
```

**Comment by HeyItsBethany3** [comment]
Thank you so much! - I downgraded the version and it worked

**Comment by rkooo567** [comment]
I will take a look at this issue. @iycheng is it possible to reproduce this if I remove version constraint on the CI?

**Comment by danieljanes** [comment]
Is Ray going to publish patch releases for older versions that include the `<=1.43` version range limiter?

**Comment by rkooo567** [comment]
Cc @iycheng for the release ^

**Comment by ericl** [comment]
@mwtian @iycheng is this fixed in the latest release?

**Comment by mwtian** [comment]
Yes, the fix (99fa8424f0e8129448cfddb394bb1a9a7829399b) has been cherry picked into 1.11.0. cc @iycheng

**Comment by rkooo567** [comment]
Btw it is not the long term fix. I think we need to mark it P1 and find the long term fix to allow us to use 1.44 (according to philipp this is the first version that works well with M1 Mac)

**Comment by pcmoritz** [comment]
For some reason, this is one of the top viewed pages on github:

<img width="472" alt="Screen Shot 2022-03-05 at 9 46 34 AM" src="https://user-images.githubusercontent.com/113316/156894460-9577ac86-44d2-4abc-9a8f-3b42062a7ccf.png">

**Comment by mcharlou** [comment]
@pcmoritz I think that's because the issue stems from grpcio 1.44, so it affected much more than just Ray. I'm not a Ray user but ended up here because I've got the same issue on my own program. The [latest issue regarding this message on the gRPC GitHub](https://github.com/grpc/grpc/issues/17253) is closed and 4 years old, so people naturally end up here.

**Comment by mwtian** [comment]
The issue should have been fixed on gRPC `master`. We can wait for the release containing the fix.

**Comment by calisro** [comment]
> @HeyItsBethany3 I just downgraded to grpcio==1.43, which is a workaround until there's a more fundamental fix.

IS this an actual work around for now?  Does this break anything?

**Comment by rkooo567** [comment]
yeah that's the actual workaround now!

**Comment by rkooo567** [comment]
Btw does anyone work on this @mwtian ?

**Comment by mwtian** [comment]
@rkooo567 We are waiting on gRPC to make a release that sorts out the issue and remove the pins.

**Comment by vakker** [comment]
I'm not sure this is over yet.

Now, with grpcio 1.46.0 (and latest Ray from master) I see a lot of these:
```sh
(PPOTrainer pid=92813, ip=10.10.4.2) E0506 14:39:06.571835791   93706 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers  
```

Which is coming directly from below the previous message, see [here](https://github.com/grpc/grpc/blob/master/src/core/lib/iomgr/fork_posix.cc#L76).

With grpcio 1.43.0 I don't see any. Seems to be the same issue, i.e. it was changed to ERROR in 1.44.0.
So, the question is the same, does this have any impact on Ray or just log clutter?

**Comment by mwtian** [comment]
@vakker this message would be useful if the Python code actually tries to fork processes. Btw for your application, is it multi-threaded and does it use fork explicitly? Is there an easy way to reproduce the issue?

**Comment by vakker** [comment]
@mwtian, yes, there's a very simple repro example:
```python
import ray
from ray import tune

ray.init()

config = {
    "env": "CartPole-v0",
    "framework": "torch",
}

stop = {"training_iteration": 10}

results = tune.run("PPO", config=config, stop=stop, verbose=1)
```

Output:
```sh
(PPOTrainer pid=3637823) 2022-05-09 09:48:24,247        INFO ppo.py:358 -- In multi-agent mode, policies will be optimized sequentially by the multi-GPU optimizer. Consider setting simple_optimizer=True if this 
doesn't work for you.                                                                                    
(PPOTrainer pid=3637823) 2022-05-09 09:48:24,247        INFO trainer.py:328 -- Current log_level is WARN. For more information, set 'log_level': 'INFO' / 'DEBUG' or use the -v and -vv flags.
== Status ==     
Current time: 2022-05-09 09:48:27 (running for 00:00:06.62)                                                                                                                                                        
Memory usage on this node: 4.7/62.7 GiB    
Using FIFO scheduling algorithm.                                                                                                                                                                                   
Resources requested: 3.0/24 CPUs, 0/2 GPUs, 0.0/36.49 GiB heap, 0.0/18.25 GiB objects (0.0/1.0 accelerator_type:G)
Result logdir: /home/user/ray_results/PPO                                                                                                                                                                         
Number of trials: 1/1 (1 RUNNING)
                                                                                                                                                                                                                   
                                                    
(PPOTrainer pid=3637823) E0509 09:48:27.555485149 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:28.284624243 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:29.011555875 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:29.738790149 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:30.466852861 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:31.195443639 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:31.922939044 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
== Status ==                 
Current time: 2022-05-09 09:48:32 (running for 00:00:11.63)
Memory usage on this node: 4.7/62.7 GiB
Using FIFO scheduling algorithm.
Resources requested: 3.0/24 CPUs, 0/2 GPUs, 0.0/36.49 GiB heap, 0.0/18.25 GiB objects (0.0/1.0 accelerator_type:G)
Result logdir: /home/user/ray_results/PPO
Number of trials: 1/1 (1 RUNNING)


(PPOTrainer pid=3637823) E0509 09:48:33.380481741 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:34.108300639 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:34.836662232 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:36.293836842 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:37.751692387 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:38.478821377 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:39.206408191 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers
(PPOTrainer pid=3637823) E0509 09:48:42.119425858 3637988 fork_posix.cc:76]           Other threads are currently calling into gRPC, skipping fork() handlers

.....
```

This is with ray 7ab19ddc3 and grpcio 1.43.0.

**Comment by mwtian** [comment]
@vakker I reproduced this log message with grpcio `1.46.0`, but it appeared only once. Since the condition that triggered the logging likely exists in older versions of `grpcio`, this should not affect the output of Ray program. Pinning `grpcio` to `1.43.0` or an earlier version would avoid the excessive logging.

**Comment by mwtian** [comment]
I just re-pinned grpcio to <= 1.43.0 in Ray via https://github.com/ray-project/ray/commit/502c3e132d27ce5106fe98ecde8deacbdde57e6a

**Comment by vrazdalovschi** [comment]
Is it possible to update to 1.47+? We need 1.47+, because previous versions have issues with mac m1 wheels and need to do unpleasant interactions to support it. Thank you!

**Comment by mwtian** [comment]
@vrazdalovschi It seems the logspam fix is [available in gRPC 1.48](https://github.com/grpc/grpc/commit/e8ca82b9a4374c8f87c2d190509c39055deda42a), so we can try using that. Btw only Python `grpcio` should need to be upgraded. I can send out a PR.

**Comment by rkooo567** [comment]
cc @scv119 for answers ^?
